server/Server/OpenLibrary/OpenLibraryClientDecrator.cs

```diff
@@ -0,0 +1,30 @@
+using Microsoft.EntityFrameworkCore;
+using Server.Domain;
+using Server.Domain.Models;
+
+namespace Server.OpenLibrary;
+
+public class OpenLibraryClientDecorator(BookRepoContext context, OpenLibraryClient client)
+    : IOpenLibraryCient
+{
+    public async Task<Book?> GetBook(string isbn, CancellationToken cancellationToken)
+    {
+        if ((await context.Books.FindAsync([ isbn ], cancellationToken)) is { } book)
```

Comment:
We should add a `Last Updated` time onto `Book` then if x amount of time has passed we can call the endpoint to update the book data